### PR TITLE
Feature/handler ping

### PR DESCRIPTION
**Pull request by AgwaB**
resolved: #[72]

details:

-Implement New function more detail
-pb message is not implemented exactly yet so implement ping handler simply
-Change message seq -> id

-I make a mistake in #82 commit, so close it and PR it again.

### CODE REVIEW

**Comment by hihiboss** [comment]
The commits are right?
Also, please fix the conflicting files!

File: swim.go [hunk 1]
```diff
@@ -45,24 +52,53 @@ type SWIM struct {
 	// Currently connected memberList
 	memberMap *MemberMap
 
+	messageEndpoint *MessageEndpoint
+
```
Comment (junbeomlee) [inline swim.go]: pbkstore already exists!

**Comment by zeroFruit** [comment]
i think handlePing need test cases

File: swim.go [hunk 2]
```diff
@@ -58,17 +64,40 @@ type SWIM struct {
...
+func messageEndpointFactory(config *Config, messageEndpointConfig MessageEndpointConfig, messageHandler MessageHandler, awareness *Awareness) *MessageEndpoint {
+	packetTransportConfig := PacketTransportConfig{
+		BindAddress: config.BindAddress,
+		BindPort:    config.BindPort,
+	}
+
+	packetTransport, err := NewPacketTransport(&packetTransportConfig)
+	if err != nil {
+		log.Panic(err)
+	}
```
Comment (zeroFruit) [inline swim.go:91]: log with ILogger would be better!

File: swim_internal_test.go [hunk 3]
```diff
@@ -89,10 +96,73 @@ func TestSWIM_handlePbk(t *testing.T) {
...
+		return pb.PiggyBack{
+			Type:        pb.PiggyBack_Alive,
+			Id:          "pbk_id1",
+			Incarnation: 123,
+			Address:     "address123",
+		}, nil
+	}
+
+	mIMessageHandler := MockMessageHandler{}
+	mJMessageHandler := MockMessageHandler{}
```
Comment (zeroFruit) [inline swim_internal_test.go]: mJMessageHandler don't need, swim (line 116) work as message handler for mJ

File: swim_internal_test.go [hunk 4]
```diff
@@ -89,10 +96,73 @@ func TestSWIM_handlePbk(t *testing.T) {
...
+	mJMessageHandler.handleFunc = func(msg pb.Message) {
+		// check whether msg.Payload type is *pb.Message_Ack
+		assert.NotNil(t, msg.Payload.(*pb.Message_Ping))
+		assert.Equal(t, msg.PiggyBack.Address, "address123")
+		assert.Equal(t, msg.PiggyBack.Incarnation, uint32(123))
+		assert.Equal(t, msg.PiggyBack.Id, "pbk_id1")
+
+		ack := pb.Message{Id: msg.Id, Payload: &pb.Message_Ack{Ack: &pb.Ack{}}, PiggyBack: &pb.PiggyBack{}}
+		mJ.Send("127.0.0.1:11140", ack)
+	}
```
Comment (zeroFruit) [inline swim_internal_test.go]: this `handleFunc` would not execute

**Review by zeroFruit — APPROVED** [review]

**Pull request merged**